Mathlib/GroupTheory/CosetCover.lean

```diff
@@ -370,19 +370,31 @@ variable {k E : Type*} [DivisionRing k] [Infinite k] [AddCommGroup E] [Module k
     {s : Finset (Subspace k E)}
 
 /- A vector space over an infinite field cannot be a finite union of proper subspaces. -/
-theorem Subspace.biUnion_ne_univ_of_ne_top (hs : ∀ p ∈ s, p ≠ ⊤) :
+theorem Subspace.biUnion_ne_univ_of_top_nmem (hs : ⊤ ∉ s) :
     ⋃ p ∈ s, (p : Set E) ≠ Set.univ := by
   intro hcovers
   have ⟨p, hp, hfi⟩ := Submodule.exists_finiteIndex_of_cover hcovers
   have : Finite (E ⧸ p) := AddSubgroup.finite_quotient_of_finiteIndex _
-  have : Nontrivial (E ⧸ p) := Submodule.Quotient.nontrivial_of_lt_top p (hs p hp).lt_top
+  have : Nontrivial (E ⧸ p) :=
+    Submodule.Quotient.nontrivial_of_lt_top p (ne_of_mem_of_not_mem hp hs).lt_top
   have : Infinite (E ⧸ p) := Module.Free.infinite k (E ⧸ p)
   exact not_finite (E ⧸ p)
 
 /- A vector space over an infinite field cannot be a finite union of proper subspaces. -/
-theorem Subspace.exists_eq_top_of_biUnion_eq_univ (hcovers : ⋃ p ∈ s, (p : Set E) = Set.univ) :
-    ∃ p ∈ s, p = ⊤ := by
+theorem Subspace.top_mem_of_biUnion_eq_univ (hcovers : ⋃ p ∈ s, (p : Set E) = Set.univ) :
+    ⊤ ∈ s := by
   contrapose! hcovers
-  exact Subspace.biUnion_ne_univ_of_ne_top hcovers
+  exact Subspace.biUnion_ne_univ_of_top_nmem hcovers
+
+@[deprecated (since := "2024-10-29")]
+alias Subspace.biUnion_ne_univ_of_ne_top := Subspace.biUnion_ne_univ_of_top_nmem
+@[deprecated (since := "2024-10-29")]
+alias Subspace.exists_eq_top_of_biUnion_eq_univ := Subspace.top_mem_of_biUnion_eq_univ
+
+theorem Subspace.exists_eq_top_of_iUnion_eq_univ {ι} [Finite ι] {p : ι → Subspace k E}
```

Comment:
```suggestion
theorem Subspace.exists_eq_top_of_iUnion_eq_univ {ι : Type*} [Finite ι] {p : ι → Subspace k E}
```
or `Sort*`?

Reply:
`Finite` allows `Sort*`. Leaving it unspecified has the advantage that you don't need to worry about this.